packages/macro/src/macroJsx.ts

```diff
@@ -87,11 +87,15 @@ export default class MacroJSX {
         }
       }
     } else {
+      // Quoted JSX attributes use XML-style escapes instead of JavaScript-style escapes.
+      // This means that <Trans id="Say \"hi\"!" /> is invalid, but <Trans id={"Say \"hi\"!"} /> works.
+      // We could consider removing the condition here and always wrap in a jsxExpressionContainer.
```

Comment:
I tested out removing this conditional and didn’t find any regressions. It might fix other cases that the double quoting issue. 

Let me know if we should remove the conditional here.